src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.FormatInfo.cs

```diff
@@ -50,8 +50,68 @@ internal static int GetFormatInfoForMountPoint(string name, out DriveType type)
 
+        /// <summary>
+        /// Retrieves format information for the specified mount point.
+        /// </summary>
+        /// <remarks>
+        /// This method uses a two-step approach to retrieve filesystem information:
+        /// 1. On Linux systems, it first attempts to read from `/proc/self/mountinfo`.
+        /// 2. If step 1 fails or on non-Linux systems, it falls back to a P/Invoke call.
+        ///
+        /// The `/proc/self/mountinfo` approach is preferred on Linux because
+        /// it's more reliable when procfs is available and functioning correctly.
+        ///
+        /// For other systems:
+        /// - SunOS and similar: procfs provides a binary interface, not suitable for this method.
+        /// - macOS: procfs is not available.
+        /// - FreeBSD: procfs is optional and not enabled by default.
+        ///
+        /// The method uses try/catch blocks to ensure robustness, even though `/proc/self/mountinfo`
+        /// should be specific to each process according to kernel documentation.
+        /// </remarks>
+        /// <param name="name">The mount point name to query.</param>
+        /// <param name="format">Output parameter for the filesystem format.</param>
+        /// <param name="type">Output parameter for the drive type.</param>
+        /// <returns>0 if successful, otherwise an error code.</returns>
         private static unsafe int GetFormatInfoForMountPoint(string name, out string format, out DriveType type)
         {
+#if TARGET_LINUX
+            try
+            {
+                const string mountInfoFilePath = "/proc/self/mountinfo";
+                var mountInfoFileContent = File.ReadAllLines(mountInfoFilePath);
+                foreach (var line in mountInfoFileContent)
+                {
+                    StringParser parser = new StringParser(line, ' ');
+            
+                    // Skip fields we don't care about (Fields 1-4)
+                    parser.MoveNext(); // Skip Mount ID
+                    parser.MoveNext(); // Skip Parent ID
+                    parser.MoveNext(); // Skip Major:Minor
+                    parser.MoveNext(); // Skip Root
+            
+                    // Get the mount point (Field 5)
+                    string mountPoint = parser.MoveAndExtractNext();
+            
+                    // Skip to the separator which is end of optional fields (Field 8)
+                    while (parser.MoveAndExtractNext() != "-")
+                    {
+                    }
+            
+                    // Get filesystem type (Field 9)
+                    string filesystemType = parser.MoveAndExtractNext();
+            
+                    if (mountPoint == name)
+                    {
+                        format = filesystemType;
+                        type = GetDriveType(filesystemType);
+                        return 0;
+                    }
+                }
+            }
+            catch { /* ignored */ }
+#endif
```

Comment:
.NET 9 split enumerator can help in making it more memory efficient. Something like this:
```suggestion
#if TARGET_LINUX
            try
            {
                using StreamReader reader = new("/proc/self/mountinfo");
                
                string rawLine;
                while ((rawLine = reader.ReadLine()) is not null)
                {
                    ReadOnlySpan<char> line = rawLine.AsSpan();
                    MemoryExtensions.SpanSplitEnumerator<char> fields = line.Split(' ');
                
                    // Skip fields we don't care about (Fields 1-4)
                    fields.MoveNext(); // Skip Mount ID
                    fields.MoveNext(); // Skip Parent ID
                    fields.MoveNext(); // Skip Major:Minor
                    fields.MoveNext(); // Skip Root

                    if (!MemoryExtensions.Equals(line[fields.Current], name, StringComparison.Ordinal)) continue;

                    // Skip to the separator which is end of optional fields (Field 8)
                    while (fields.MoveNext() && !MemoryExtensions.Equals(line[fields.Current], "-", StringComparison.Ordinal));

                    fields.MoveNext();
                    format = line[fields.Current].ToString();

                    fields.MoveNext();
                    type = GetDriveType(line[fields.Current].ToString());
                    
                    return 0;
            }
            catch { /* ignored */ }
#endif
```

Reply:
Sorry, I accidentally pressed the accept button in Github. I've reverted the changes.

Your code isn't working correctly because "fields.Current" returns only a single character instead of the entire mount point string.

Btw, I used the string parser based on a suggestion from @tmds, and other similar functionalities use the same class.

Reply:
“Something like that” suggestion imply confirming that it’s working before merging. I actually tested it in Linux machine in a standalone app and it was returning the same values before and after.

> Your code isn't working correctly because "fields.Current" returns only a single character instead of the entire mount point string

.Current returns Range and it is only of length 1 for a single character.

Reply:
I know how it works :)
But in your code, we skip the first 4 fields and start reading, which means we read the root instead of the mount point, I will add your suggestion and push a fix for that. I'm not sure how it works for you! Maybe you can just check against / drive in your test case.

Reply:
It was a quick test on wsl with daily build while I was on a tram on my way to work, which is why I was relying on your double-checking. 😅
```c#
Console.WriteLine(GetFormatInfoForMountPoint());

static int GetFormatInfoForMountPoint()
{
    using StreamReader reader = new("/proc/self/mountinfo");

    string rawLine;
    while ((rawLine = reader.ReadLine()) is not null)
    {
        ReadOnlySpan<char> line = rawLine.AsSpan();
        MemoryExtensions.SpanSplitEnumerator<char> fields = line.Split(' ');

        // Skip fields we don't care about (Fields 1-4)
        fields.MoveNext(); // Skip Mount ID
        fields.MoveNext(); // Skip Parent ID
        fields.MoveNext(); // Skip Major:Minor
        fields.MoveNext(); // Skip Root

        string name = "foo";
        //                    if (line[fields.Current] != name) continue;
        //                    fields.MoveNext();
        while (fields.MoveNext() && !MemoryExtensions.Equals(line[fields.Current], "-", StringComparison.Ordinal)) ;// {Console.Wr>                                                                                                                    //MemoryExtens>                                                                                                                    //            >        fields.MoveNext();

        string format = line[fields.Current].ToString();
        Console.WriteLine(format);
        //return 0;
    }

    return 1;
}
```

Reply:
fine, I pushed the fix for that. also refactored a bit for the check parts.